src/game/logic/object/objecttypes.cpp

```diff
@@ -44,3 +48,80 @@ void ObjectTypes::Xfer_Snapshot(Xfer *xfer)
         }
     }
 }
+
+Utf8String ObjectTypes::Get_Nth_In_List(unsigned int i) const
+{
+    if (i > Get_List_Count()) {
+        return Utf8String::s_emptyString;
+    } else {
+        return m_objectTypes[i];
+    }
+}
+
+void ObjectTypes::Add_Object_Type(const Utf8String &type)
+{
+    if (!Is_In_Set(type)) {
+        m_objectTypes.push_back(type);
+    }
+}
+
+void ObjectTypes::Remove_Object_Type(const Utf8String &type)
+{
+    if (Is_In_Set(type)) {
+        m_objectTypes.erase(std::find(m_objectTypes.begin(), m_objectTypes.end(), type));
+    } else {
+        captainslog_dbgassert(
+            false, "Attempted to remove '%s' from '%s', but it wasn't there.", type.Str(), m_listName.Str());
+    }
+}
+
+const Utf8String *ObjectTypes::Get_List_Name() const
```

Comment:
Returns pointer, not reference for sure?

Reply:
fixed.